### ISSUE TITLE
User not able to connect https://rarible.com/

### ISSUE THREAD

**Issue by HEagle18** [brave/brave-browser#23886]
## Description 

https://community.brave.com/t/bug-brave-wallet-not-connecting-to-rarible/409785 customer mentioned it here. 

I tried on 1.40.107 and on 1.42.57 and couldn't signin either. Wallet setting is `Brave Wallet only`

## Steps to Reproduce

   1. visit https://rarible.com/connect
   2. try connect brave wallet by clicking on metamask option
   3. 

## Actual result:

<img width="690" alt="Screenshot 2022-07-05 at 21 28 41" src="https://user-images.githubusercontent.com/12224304/177402057-eb136289-eda2-4d1d-8d01-d146435ffcd2.png">
Screenshot is of attempt on 1.40.107

## Reproduces how often: 

all the time

## Brave version (brave://version info)

1.40.107 and on 1.42.57

## Version/Channel Information:

- Can you reproduce this issue with the current release? yes
- Can you reproduce this issue with the beta channel? n/a
- Can you reproduce this issue with the nightly channel? yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? n/a
- Does the issue resolve itself when disabling Brave Rewards? n/a
- Is the issue reproducible on the latest version of Chrome? no

**Comment by onyb** [comment]
Found several issues with Rarible Connect.

1. Even if `window.ethereum` is available via Brave Wallet, if Metamask is not installed, clicking on the fox takes me to https://metamask.io/download.
    <img width="250" alt="Screenshot 2022-07-08 at 15 21 46" src="https://user-images.githubusercontent.com/3684187/177966951-efb72967-986a-4ec3-ba7f-ccd72a3b2da4.png">

2. Having Metamask installed but Brave Wallet set as the default, I randomly got an error saying that the connected wallet does not support switching networks on the fly (which is not true). The error disappeared after disconnecting/reloading a few times.

Once the intermittent error above disappeared, I was able to successfully connect and signup. I was on Brave Browser 1.40.x.
